.github/workflows/run_server_ci.yml

```diff
@@ -4,10 +4,13 @@
 name: Run Server Code CI
 
 on:
+  push:
+    branches: ["main"]
+    paths:
+      - 'src/server/**'
+      - '.github/workflow/**'
+      - 'ci/**'
```

Comment:
Also add the path to ui code since we are using the latest ui code to start the server with (see comment above)